### PR TITLE
[IMP] mail: `ChatterTopbar` remove inconsistent style

### PR DESCRIPTION
**Pull request by Brieuc-brd**
Prior this commit, there were several inconsistencies with the corners
of `ChatterTopbar` buttons. Some of them had rounded corners, others did
not.

After this commit, the `ChatterTopbar` buttons have the same consistent
style: rounded in Community and square in Enterprise.

task-2781992

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/static/src/components/chatter_topbar/chatter_topbar.scss [hunk 1]
```diff
@@ -19,7 +22,8 @@
 // Style
 // ------------------------------------------------------------------
 
-.o_ChatterTopbar_button {
+button.o_ChatterTopbar_button,
+.o_ChatterTopbar_button button {
```
Comment (alexkuhn) [inline addons/mail/static/src/components/chatter_topbar/chatter_topbar.scss:26]: I guess it's to target `.o_FollowerListMenu_buttonFollowers` and `.o_FollowerListMenu_buttonFollowers`?

We don't want to use "whitespace" operator in CSS rules, because this forces us to deal with CSS specificity (and reduces control on styling). Also no tag like `button` because this leaks technical details from XML, and it has higher specificity than classnames.

By lack of control, I mean that you could target more things than you really want. Which actually is the case here:
`.o_ChatterTopbar_button` would also consider `.o_Follower_editButton` from `<FollowerListMenu>` component (that itself mounts `<Follower>` conditionally). I'm pretty sure you did not want to target it, and it breaks style.

A proper way would be to pass the info to the sub-component, so that itself has special styling on these buttons.
For example, pass `isChatterButton="true"` as prop to `<FollowerButton>` and `<FollowerListMenu>` in the template, and let these components add `o_ChatterTopbar_button` in their buttons.

**Review by alexkuhn — CHANGES_REQUESTED** [review]

File: addons/mail/static/src/components/chatter_topbar/chatter_topbar.scss (hunk 1 above)
Comment (Brieuc-brd) [reply addons/mail/static/src/components/chatter_topbar/chatter_topbar.scss]: Indeed, I didn't see the impact on the `o_Follower_editButton`, sorry ^^'.

As you suggested, I added the prop to control where the `o_ChatterTopbar_button` should be placed.

What do you think about it?

Comment (alexkuhn) [reply addons/mail/static/src/components/chatter_topbar/chatter_topbar.scss]: This is good!

Just a bit strange to have `ChatterTopbar/button` inside `ChatterTopbar/button` conceptually.
I think it could be possible to distinguish them better.
Anyway, it's a quite small problem, the current code is good enough.

=====

Note that in the near future, this props will be replaced by something else, so don't be surprised to no longer see this props eventually. Here are some explanations on why we will do this:

Passing contextual info in templates by means of props is good, but we can do even better:
The code of discuss relies heavily on a modelling of the state of the application, and components should just be dumb and reflect state in the UI.
The small issue here is that the component is smarter than it should, and the logic is not in the modelling.
Ideally, the contextual info should just be in model.
The template of components in this PR do not yet follow properly this design, and we are making a pass on all these components to clean this. When all is done, contextual info is no longer passed with props, but instead model fields.

so TL;DR: passing props is good for now, but don't be surprised to no longer see this props in the near future, because it's been turned to a "model field", so that it's properly incorporated in the modelling of the discuss app.

**Review by alexkuhn — APPROVED** [review]
Will r-plus when required PR is deployed.

**Comment by Brieuc-brd** [comment]
Branch rebased

File: addons/mail/static/src/components/follow_button/follow_button.xml [hunk 2]
```diff
@@ -3,9 +3,9 @@
 
     <t t-name="mail.FollowButton" owl="1">
         <t t-if="thread">
-            <div class="o_FollowButton" t-attf-class="{{ className }}" t-ref="root">
+            <div class="o_FollowButton" t-attf-class="{{ className }} {{ props.isChatterButton ? '' : 'o_ChatterTopbar_button' }}" t-ref="root">
```
Comment (alexkuhn) [inline addons/mail/static/src/components/follow_button/follow_button.xml:6]: I think this is incorrect? You don't set `.o_ChatterTopbar_button` here because it's set on the `<button>` below, but when `FollowButton` is not in chatter, it doesn't make sense to set `o_ChatterTopbar_button` on root node.

**Review by alexkuhn — APPROVED** [review]
@robodoo r+
